pkg/trojan/message.go

```diff
@@ -163,32 +163,40 @@ func (m *Message) toChunk(targets Targets, span []byte) (swarm.Chunk, error) {
 		return nil, err
 	}
 
-	// hash chunk fields with different nonces until an acceptable one is found
-	for start := time.Now(); ; {
-		s := append(append(span, nonce...), b...) // serialize chunk fields
-		hash1, err := hashBytes(s)
-		if err != nil {
-			return nil, err
-		}
+	done := make(chan bool, 1)
+	var hash, s []byte
+	go func() {
+		// mining operation: hash chunk fields with different nonces until an acceptable one is found
+		for {
+			s = append(append(span, nonce...), b...) // serialize chunk fields
+			hash, err = hashBytes(s)
+			if err != nil {
+				return
+			}
```

Comment:
If this function would return an error, not that it will not be visible, but also this function will wait on MinerTimeout in order to return, which is not needed.

Reply:
good catch. Fixed